src/Microsoft.Azure.WebJobs.ServiceBus/EventHubs/EventHubLeaseFactory.cs

```diff
@@ -0,0 +1,133 @@
+using Microsoft.WindowsAzure.Storage.Table;
+
+namespace Microsoft.Azure.WebJobs.ServiceBus
+{
+    // EventProcessorHost requires a container name that the multiple host instances use to coordinate with each other and create leases. 
+    // An event hub is described by an (namespace name, event hub path name). The mapping from the event hub to the container name
+    // must be a one-to-one injective function such that:
+    // 1. it's deterministic: when we scale out ot N separate instances, they must all compute the same container name for the same event hub. 
+    // 2. it's collision free: different event hubs shouldn't share the same container. So we can't just use a hash. 
+    //  Note that the event hub name is much more expressive than the storage container name, so we can't do a stateless collision-free transform. 
+    // Hence we use a lookaside storage table to save the transform. 
+    internal class EventHubLeaseFactory
+    {
+        // Name of Azure table that maps "eventHub" --> container name. 
+        private const string TableName = "AzureFunctionsEventHubTable";
+             
+        // Get container name. Lease name that multiple eventhub instances use to coordinate with each other. 
+        public static string GetContainerName(string eventHubName, string eventHubNamespace, string storageConnectionString)
+        {
+            CloudStorageAccount account = CloudStorageAccount.Parse(storageConnectionString);
+            var client = account.CreateCloudTableClient();
+            var table = client.GetTableReference(TableName);
+
+            string pk = EscapeStorageKey(eventHubNamespace);
+            string rk = EscapeStorageKey(eventHubName);
+
+            // Create a retrieve operation that takes a customer entity.
+            var retrieveOperation = TableOperation.Retrieve<EventHubTableEntity>(pk, rk);
+
+            // Execute the retrieve operation.
+            Exception lastException = null;
+            for (int i = 0; i < 3; i++)
+            {
+                try
+                {
+                    var retrievedResult = table.Execute(retrieveOperation);
+
+                    if (retrievedResult.Result != null)
+                    {
+                        // Common case when we're looking up an existing event hub. 
+                        var entity = (EventHubTableEntity)retrievedResult.Result;
+                        return entity.ContainerName;
+                    }
+                    else
+                    {
+                        // Storage container, restrict to: [a-z0-9\-]{3,63}, no consecutive dashes, start&end with alpha
+                        // Guid is 32 chars long and ToString() yields lowercase.  
+                        string newName = "eventhubs-" + Guid.NewGuid().ToString("N");
+
+                        // Not found. We'll create one. 
+                        var entity = new EventHubTableEntity { PartitionKey = pk, RowKey = rk, ContainerName = newName };
```

Comment:
Do we really need a separate unique container per event hub? I'd expect that within the container, they're managing the uniqueness of their lease IDs. Why can't we just specify a single simple top level container like "eventhub-leases" and let them manage it? I can't imagine that they expect all of their users to have to replicate logic like you've done here - if that is required, it should be centralized in their SDK.